net/HttpRequest.hpp

```diff
@@ -1168,6 +1168,11 @@ class Session final : public ProtocolHandlerInterface
     /// regardless of the reason (error, timeout, completion).
     void setFinishedHandler(FinishedCallback onFinished) { _onFinished = std::move(onFinished); }
 
+    /// The onConnectFail callback handler signature.
+    using ConnectFailCallback = std::function<void()>;
```

Comment:
Wouldn't it be helpful to pass in the `http::Session` in question? Can be captured in a lambda, but seems cleaner that we pass it as context. (C.f. `FinishedCallback` on line 1164.)